frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboard.tsx

```diff
@@ -259,14 +273,30 @@ class PublicOrEmbeddedDashboardInner extends Component<PublicOrEmbeddedDashboard
           })}
           loading={!dashboard}
         >
-          {() =>
-            dashboard ? (
+          {() => {
+            if (!dashboard) {
+              return null;
+            }
+
+            if (!dashboardHasCards || !tabHasCards) {
+              return (
+                <DashboardEmptyStateWithoutAddPrompt
+                  isNightMode={isNightMode}
+                />
+              );
+            }
+
+            return (
               <DashboardContainer>
                 <DashboardGridConnected
                   dashboard={assoc(dashboard, "dashcards", visibleDashcards)}
                   isPublicOrEmbedded
-                  mode={PublicMode as unknown as Mode}
-                  selectedTabId={this.props.selectedTabId}
+                  mode={
+                    navigateToNewCardFromDashboard
```

Comment:
I'm not a fan of components figuring out stuff based on props like this, we have many places where components "down the road" have logic to do different things based on the props passed and it becomes complex easy.
`PublicOrEmbeddedDashboard` is used in 3 different ways: static embed, public sharing, and now from the sdk.
Should we add a `mode` prop to it and just pass it down?
We could make it require certain props based on the mode too, so that for example `navigateToNewCardFromDashboard` is required when `mode=sdk`